js/src/annotations/simpleASEndpoint.js

```diff
@@ -0,0 +1,188 @@
+/*
+* Edited version of https://github.com/IIIF/mirador/blob/9e3c6bbb894e044d01ad51aae1b70309939de5a9/js/src/annotations/catchEndpoint.js 
+* This module tries to store the annotation as is in a RDF store but some fiddeling is required. Fidles are:
+* 
+* - delete annotation fails if id has a / in it so have to send sanatised ids to mirador
+* - mirador requires an endpoint variable in the annotation pointing to this class.
+* 
+* Note: this endpoint doesn't currently support authentication, just returns allow all
+* 
+* All Endpoints need to have at least the following:
+* annotationsList - current list of OA Annotations
+* dfd - Deferred Object
+* init()
+* search(uri)
+* create(oaAnnotation, returnSuccess, returnError)
+* update(oaAnnotation, returnSuccess, returnError)
+* deleteAnnotation(annotationID, returnSuccess, returnError) (delete is a reserved word)
+* TODO:
+* There is a bug in that if you create an annotation and then delete it (without moving pages) then click either the write annotation button 
+* or try to create a new annotation the deleted annotation re-appears. Changing pages fixes the issue as the annoation is delete from the annotation store
+*
+*/
+(function($){
+
+  $.SimpleASEndpoint = function(options) {
+
+    jQuery.extend(this, {
+      token:     null,
+      // prefix:    'annotation', /**/
+      uri:       null,
+      url:		  options.url,
+      dfd:       null,
+      annotationsList: [],        //OA list for Mirador use
+      idMapper: {} // internal list for module use to map id to URI
+    }, options);
+
+    this.init();
+  };
+
+  $.SimpleASEndpoint.prototype = {
+    //Any set up for this endpoint, and triggers a search of the URI passed to object
+    init: function() {
+      this.catchOptions = {
+        user: {
+          id: this.userid, 
+          name: this.username
+        },
+        permissions: {
+          'read':   [],
+          'update': [this.userid],
+          'delete': [this.userid],
+          'admin':  [this.userid]
+        }
+      };
+      this.search(this.uri);
+    },
+
+    //Search endpoint for all annotations with a given URI
+    search: function(uri) {
```

Comment:
@aeschylus The biggest thing that needs to change is the search function will be passed JSON object of options, which includes the uri (which affects line 71).  This function can also take success and error callbacks, but not required.  See also [the catch adaptor in master](https://github.com/IIIF/mirador/blob/master/js/src/annotations/catchEndpoint.js#L63)